README.md

```diff
@@ -1,148 +1,206 @@
+As a digital service provider, you should first [register yourself with The Green Web Foundation](https://www.thegreenwebfoundation.org/green-web-check/register/), and provide evidence of your green claims.
 
+1. **Register with The Green Web Foundation** \
+   As a digital service provider, you should first [register yourself with The Green Web Foundation](https://www.thegreenwebfoundation.org/green-web-check/register/), and provide evidence of your green claims.
+2. **Create a carbon.txt file for your organisation** \
+   Create a carbon.txt file for your organisation. There is a guide to the expected syntax below. We recommend that you host this file either at the root of your domain, or in a `.well-known` folder. \
+   For example: _<https://www.my-org.com/carbon.txt>_ or \
+   _<https://www.my-org.com/.well-known/carbon.txt>_
 
-If I host all the bits I need with one provider, like Krystal, might have a `carbon.txt` file at the root of my page, and it might look like this.
+#### Carbon.txt file syntax for digital service providers
 
-_This is all a draft - please, please file an issue to outline what you would need to see to make this something you could support_
+The carbon.txt syntax is written in TOML. Below is an example of what a carbon.txt file might look like for a digital service provider.
 
-```
+```toml
 [upstream]
-krystal.co.uk
-
-
+providers = [
+    # An array of providers my-org.com is using to deliver our service
+    { domain='cloud.google.com', service = 'infrastructure' },
+    { domain='aws.amazon.com', service = 'infrastructure' }
+]
+
+[org]
+credentials = [
+    # Optional.
+    # An array of documents that point to evidence of green claims made by my-org.com.
+    { domain='my-org.com', doctype = 'sustainability-page', url = 'https://my-org.com/our-climate-record'}
+]
 ```
 
-If a hosting company like Krystal didn't run their own datacentres, their carbon.txt file might look like this:
+_An example of a carbon.txt file for a digital service provider._
 
-```
-[upstream]
-www.netwisehosting.co.uk
-```
+#### For digital service providers with multiple domains
 
-If Netwise ran _all_ their datacentres on renewable power, all they might need to do is list their one provider like so:
+It is to be expected that one digital service provider might own and provide services through multiple domains. In this case, you can maintain a single carbon.txt at one domain (e.g. _<https://my-org.com/.well-known/carbon.txt>_) and refer other domains to that single source of truth. To do this:
 
-```
-Ecotricity Group Ltd
-```
+1. Complete the steps above to register as a provider, and create a carbon.txt file.
+2. **Generate a public key for your account** \
+   To do this, use the Observable Notebook linked below: \
+   [https://observablehq.com/d/17c922a33a5d6867](https://observablehq.com/d/17c922a33a5d6867) \
+   (Note: We understand this is solution is not ideal & are working on a nicer way to generate secrets)
```

Comment:
Now that the API doc page shows proper responses in production, I really think it's simpler to add an endpoint there to generate a hash from a provided domain, when folks are signed in with the user credentials.

Thank you for making this explicit that this won't be a thing we use for long.